central/authprovider/datastore/datastore_impl.go

```diff
@@ -29,12 +29,31 @@ func (b *datastoreImpl) GetAllAuthProviders(ctx context.Context) ([]*storage.Aut
 	return b.storage.GetAll(ctx)
 }
 
+func (b *datastoreImpl) GetAuthProvidersFiltered(ctx context.Context,
+	filter func(provider *storage.AuthProvider) bool) ([]*storage.AuthProvider, error) {
+	if err := sac.VerifyAuthzOK(accessSAC.ReadAllowed(ctx)); err != nil {
+		return nil, err
+	}
+	// TODO(ROX-XXXXX): The store currently doesn't provide a Walk function. This is mostly due to us supporting the
```

Comment:
Let's replace `XXXXX` with actual JIRA number

Reply:
Will do once JIRA maintenance window is over.